src/main/java/org/opensrp/service/StockService.java

```diff
@@ -231,22 +230,37 @@ public CsvBulkImportDataSummary convertandPersistInventorydata(List<Map<String,
 			failedRecordSummaries.add(failedRecordSummary);
 		}
 
-		if (failedRecordSummaries.size() == 0) {
-			for (Map<String, String> csvdata : csvStocks) {
-				try {
-					rowCount++;
-					inventory = createInventoryObject(csvdata);
-					addInventory(inventory, userName);
-					rowsProcessed++;
-				}
-				catch (Exception e) {
-					failedRecordSummary = new FailedRecordSummary();
-					List<String> validationError = new ArrayList<>();
-					failedRecordSummary.setRowNumber(rowCount);
-					validationError.add("Unknown error occurred");
-					failedRecordSummary.setReasonOfFailure(validationError);
-					failedRecordSummaries.add(failedRecordSummary);
-				}
+		csvBulkImportDataSummary.setFailedRecordSummaryList(failedRecordSummaries);
+		csvBulkImportDataSummary.setNumberOfCsvRows(totalRows);
+		csvBulkImportDataSummary.setNumberOfRowsProcessed(rowsProcessed);
+		return csvBulkImportDataSummary;
+
+	}
+
+	public CsvBulkImportDataSummary convertandPersistInventorydata(List<Map<String, String>> csvStocks, String userName) {
+
+		int rowCount = 0;
+		CsvBulkImportDataSummary csvBulkImportDataSummary = new CsvBulkImportDataSummary();
+		List<FailedRecordSummary> failedRecordSummaries = new ArrayList<>();
+		Inventory inventory;
+		Integer totalRows = csvStocks.size();
+		Integer rowsProcessed = 0;
+		FailedRecordSummary failedRecordSummary;
+
+		for (Map<String, String> csvdata : csvStocks) {
```

Comment:
shouldn't we validate again and not trust that validation has been done?

Reply:
As per the mockups, validation is the first step and once validation is successful then the user confirms to actually process/persist the data. @githengi

Reply:
since this is exposed via an API, the import should still validate before saving and not trust that validation was done. This is because we are not using a session to track if validation was done

Reply:
Okay noted. I will update!